nengo/spa/vocab.py

```diff
@@ -328,18 +334,33 @@ def transform_to(self, other, keys=None):
             terms in this list that do not exist in the Vocabularies will
             be created.
         """
-        if keys is None:
-            keys = list(self.keys)
-            for k in other.keys:
-                if k not in keys:
-                    keys.append(k)
-
-        t = np.zeros((other.dimensions, self.dimensions), dtype=float)
-        for k in keys:
-            a = self[k].v
-            b = other[k].v
-            t += np.outer(b, a)
-        return t
+        # If the parent vocabs of self and other are the same, then no
+        # transform is needed between the two vocabularies, so return an
+        # identity matrix.
+        my_parent = self if self.parent is None else self.parent
+        other_parent = other if other.parent is None else other.parent
+
+        if my_parent == other_parent:
```

Comment:
Is short-circuiting this check the only reason to keep the parent links around? If so, I strongly suggest we remove the parent references. They can cause lots of problems down the road, and in the best case it's an example of premature optimization. Does this function really take that much time?